File: app/admin/service/internal/biz/organization.go
package biz

import (
	"context"
	"time"

	"github.com/go-kratos/kratos/v2/log"
	"github.com/hypercoze/kratos-admin/pkg/utils/uuid"
)

type Organization struct {
	ID          string    `json:"id"`          // 部门ID
	ParentID    string    `json:"parentID"`    // 父部门ID，空表示根节点
	Name        string    `json:"name"`        // 部门名称
	Code        string    `json:"code"`        // 部门code
	Weight      int       `json:"weight"`      // 权重
	Status      string    `json:"status"`      // 状态：enabled=启用 disabled=禁用
	LeaderName  string    `json:"leaderName"`  // 负责人姓名
	LeaderPhone string    `json:"leaderPhone"` // 联系电话
	LeaderEmail string    `json:"leaderEmail"` // 邮箱
	CreatedAt   time.Time `json:"createdAt"`   // 创建时间
	UpdatedAt   time.Time `json:"updatedAt"`   // 更新时间

	Children []*Organization `json:"children"`
}

type OrganizationListResult struct {
	Name   string `form:"name" query:"name"`
	Code   string `form:"code" query:"code"`
	Status string `form:"status" query:"status"`
}

type OrganizationRepo interface {
	ListOrganization(ctx context.Context, req *OrganizationListResult) ([]*Organization, error)
	CreateOrganization(ctx context.Context, o *Organization) error
	UpdateOrganization(ctx context.Context, o *Organization) error
	DeleteOrganization(ctx context.Context, ids []string) error
	UpdateOrganizationStatus(ctx context.Context, id, status string) error
}

type OrganizationUsecase struct {
	repo OrganizationRepo
	log  *log.Helper
}

func NewOrganizationUsecase(repo OrganizationRepo, logger log.Logger) *OrganizationUsecase {
	return &OrganizationUsecase{
		repo: repo,
		log:  log.NewHelper(logger),
	}
}

// 列表权限的tree
func (uc *OrganizationUsecase) OrganizationTree(ctx context.Context, req *OrganizationListResult) ([]*Organization, error) {
	list, err := uc.repo.ListOrganization(ctx, req)
	if err != nil {
		return nil, err
	}
	tree := buildOrganizationTree(list)

	return tree, nil
}

// 创建
func (uc *OrganizationUsecase) CreateOrganization(ctx context.Context, req *Organization) (*Organization, error) {
	organization := req

	now := time.Now()
	organization.ID = uuid.GenerateXID()
	organization.CreatedAt = now
	organization.UpdatedAt = now

	err := uc.repo.CreateOrganization(ctx, organization)
	return organization, err
}

// 更新
func (uc *OrganizationUsecase) UpdateOrganization(ctx context.Context, req *Organization) (*Organization, error) {
	organization := req

	now := time.Now()
	organization.UpdatedAt = now

	err := uc.repo.UpdateOrganization(ctx, organization)
	return organization, err
}

// 删除
func (uc *OrganizationUsecase) DeleteOrganization(ctx context.Context, ids []string) error {
	return uc.repo.DeleteOrganization(ctx, ids)
}

// 更新状态
func (uc *OrganizationUsecase) UpdateOrganizationStatus(ctx context.Context, id, status string) error {
	return uc.repo.UpdateOrganizationStatus(ctx, id, status)
}

File: vue-vben-admin/apps/web-antd/src/api/system/organization.ts
import { requestClient } from '#/api/request';
import type { PaginationParams,AllResult } from '#/types/pagination'

export namespace SystemOrganizationApi {
  /** 组织机构 */
  export interface Organization {
    /** 部门ID */
    id: string;
    /** 父部门ID，空表示根节点 */
    parentId: string;
    /** 部门名称 */
    name: string;
    /** 部门编码 */
    code?: string;
    /** 权重 */
    weight: number;
    /** 状态：enabled=启用 disabled=禁用 */
    status: 'enabled' | 'disabled';
    /** 负责人姓名 */
    leaderName?: string;
    /** 联系电话 */
    leaderPhone?: string;
    /** 邮箱 */
    leaderEmail?: string;
    /** 创建时间 */
    createdAt?: string;
    /** 更新时间 */
    updatedAt?: string;
    /** 子级组织 */
    children?: Organization[];
  }

  export interface OrganizationTreeParams{
    name?: string;
    code?: string;
    status?: 'enabled' | 'disabled';
  }
}

/**
 * 获取组织机构树
 */
async function getOrganizationTreeApi(params?:SystemOrganizationApi.OrganizationTreeParams) {
  return requestClient.get<AllResult<SystemOrganizationApi.Organization>>(
    '/admin/v1/organization/tree',
    {
      params:params
    }
  );
}

/**
 * 创建组织机构
 * @param data 组织数据
 */
async function createOrganization(
  data: Omit<SystemOrganizationApi.Organization, 'id' | 'children'>,
) {
  return requestClient.post('/admin/v1/organization', data);
}

/**
 * 更新组织机构
 *
 * @param id 组织 ID
 * @param data 组织数据
 */
async function updateOrganization(
  id: string,
  data: Omit<SystemOrganizationApi.Organization, 'id' | 'children'>,
) {
  return requestClient.put(`/admin/v1/organization/${id}`, data);
}

/**
 * 更新组织机构状态
 *
 * @param id 组织 ID
 * @param status 状态
 */
async function updateOrganizationStatusApi(
  id: string,
  status: SystemOrganizationApi.Organization['status'],
) {
  return requestClient.put(`/admin/v1/organization/${id}/status`, {
    status,
  });
}

/**
 * 删除组织机构
 * @param ids 组织 ID 集合
 */
async function deleteOrganizationApi(ids: string[]) {
  return requestClient.delete('/admin/v1/organization', {
    params: {
      ids,
    },
  });
}

export {
  createOrganization,
  deleteOrganizationApi,
  getOrganizationTreeApi,
  updateOrganization,
  updateOrganizationStatusApi,
};

